tools/serverpod_cli/test/integration/serverpod_packages_version_check/load_config_test.dart

```diff
@@ -0,0 +1,55 @@
+import 'dart:io';
+
+import 'package:path/path.dart' as path;
+import 'package:serverpod_cli/src/config/config.dart';
+import 'package:test/test.dart';
+
+void main() {
+  final root = Directory.current;
+  final testAssetsPath = path.join(
+    root.path,
+    'test',
+    'integration',
+    'serverpod_packages_version_check',
+    'test_assets',
+  );
+
+  test(
+      'Given a missing server pubspec.yaml '
+      'then a ServerpodProjectNotFoundException is thrown', () async {
+    final missingPubspecDir = Directory.systemTemp.createTempSync();
+    await expectLater(
+      GeneratorConfig.load(missingPubspecDir.path),
+      throwsA(isA<ServerpodProjectNotFoundException>()),
+    );
+    await missingPubspecDir.delete();
+  });
+
+  test(
+      'Given an invalid server pubspec.yaml '
+      'then a ServerpodProjectNotFoundException is thrown', () {
+    // TODO(nielsenko):
+    // This is current behavior, but I think it should be
+    // the SourceSpanException raised by Pubspec.parse to give the user
+    // a more detailed error message.
+    expectLater(
+      GeneratorConfig.load(path.join(testAssetsPath, 'invalid_pubspec')),
+      throwsA(isA<ServerpodProjectNotFoundException>()),
+    );
+  });
+
+  test(
+      'Given a valid server pubspec.yaml'
+      ' and a missing client pubspec.yaml '
```

Comment:
Suggestion(NIT): Style fix.

```suggestion
      'Given a valid server pubspec.yaml '
      'and a missing client pubspec.yaml '
```

Reply:
Okay. It was on purpose though. It makes it easier to see the where the switch from given to then happens.

Reply:
Aah, sorry missed the intentional indentation! This makes sense! 👍 

Ignore my comment.

Reply:
But then could simply be on a single line as well right since it is part of the "Given"?(NIT...)

Reply:
I removed the prefix ' ' but kept the multiline

Reply:
Now I removed use of 'and' and multi-line as well